### ISSUE TITLE
request help: apisix cannot use CORS plug-in after upgrading from 2.0 to 2.5

### ISSUE THREAD

**Issue by suifff** [apache/apisix#4358]
### Issue description
I used docker deployment, and apisix-dashboard was upgraded from 2.1 to 2.6

### Environment

docker-compose.yml
~~~
version: "3"

services:
  apisix:
    #image: apache/apisix:2.0-alpine
    image: apache/apisix:2.5-alpine
    restart: always
    volumes:
      - ./apisix_log:/usr/local/apisix/logs
      - ./apisix_conf/config.yaml:/usr/local/apisix/conf/config.yaml:ro
    depends_on:
      - etcd
    ##network_mode: host
    ports:
      - "80:9080/tcp"
      - "443:9443/tcp"
#      - "9080:9080/tcp"
#      - "9443:9443/tcp"
    networks:
      apisix:
        ipv4_address: 172.20.5.11

  etcd:
    image: bitnami/etcd:3.4.9
    user: root
    restart: always
    volumes:
      - ./etcd_data:/etcd_data
    environment:
      ETCD_DATA_DIR: /etcd_data
      ETCD_ENABLE_V2: "true"
      ALLOW_NONE_AUTHENTICATION: "yes"
      ETCD_ADVERTISE_CLIENT_URLS: "http://0.0.0.0:2379"
      ETCD_LISTEN_CLIENT_URLS: "http://0.0.0.0:2379"
    ports:
      - "2379:2379/tcp"
    networks:
      apisix:
        ipv4_address: 172.20.5.10

  apisix-dashboard:
    container_name: apisix-dashboard
    #image: apache/apisix-dashboard:2.1
    image: apache/apisix-dashboard:2.6
    ports:
      - 9000:9000
    volumes:
      - ./apisix-dashboard/conf.yaml:/usr/local/apisix-dashboard/conf/conf.yaml
    networks:
      apisix:
        ipv4_address: 172.20.5.18


networks:
  apisix:
    driver: bridge
    ipam:
      config:
      - subnet: 172.20.0.0/16
~~~

Now CORS is no longer accessible
![image](https://user-images.githubusercontent.com/37427008/120444980-c1577600-c3ba-11eb-94ca-37de1c284b37.png)

This is the configuration of CORS 
![image](https://user-images.githubusercontent.com/37427008/120445320-14312d80-c3bb-11eb-96d0-5d59b96ff3a7.png)

**Comment by spacewander** [comment]
What is your configure data in the etcd?

**Comment by suifff** [comment]
> What is your configure data in the etcd?

@spacewander 通过etcdctl get --prefix /获取的值解析如下
~~~
	"plugins": {
		"cors": {
			"allow_credential": true,
			"allow_headers": "*",
			"allow_methods": "*",
			"allow_origins": "*",
			"allow_origins_by_regex": ["*"],
			"disable": false,
			"expose_headers": "*",
			"max_age": 5
		}
	},
	"upstream_id": "345046292138819799",
	"status": 1
}
~~~

**Comment by suifff** [comment]
![image](https://user-images.githubusercontent.com/37427008/120450876-7726c280-c3c3-11eb-90af-333494d4a80f.png)

**Comment by Firstsawyou** [comment]
Does apisix have any related error log prompts?

**Comment by suifff** [comment]
> Does apisix have any related error log prompts?

@Firstsawyou yes,I upgraded in the newly opened docker compose, and found that the routing difference is not big, but it cannot be accessed after upgrading. After upgrading, an interface is 404

![image](https://user-images.githubusercontent.com/37427008/120456107-e30b2a00-c3c7-11eb-8533-5bfee989c004.png)
![image](https://user-images.githubusercontent.com/37427008/120456300-0afa8d80-c3c8-11eb-8795-0f3bdd7de9c0.png)
![image](https://user-images.githubusercontent.com/37427008/120456374-1d74c700-c3c8-11eb-896c-f8679359b772.png)

**Comment by spacewander** [comment]
It is a known bug: https://github.com/apache/apisix/pull/4056

**Comment by suifff** [comment]
> It is a known bug: #4056

2.6 doesn't work

![image](https://user-images.githubusercontent.com/37427008/120472875-025e8300-c3d9-11eb-8176-b345989f8c7b.png)

**Comment by spacewander** [comment]
Can you provide a pasteable configuration so we can try it?
Also, any interesting thing in the error.log?

**Comment by suifff** [comment]
> Can you provide a pasteable configuration so we can try it?
> Also, any interesting thing in the error.log?

You may not be able to reproduce it. It's normal for me to create a new route. Is there a cache problem

**Comment by spacewander** [comment]
So it can't be reproduced now after restarting APISIX?

**Comment by suifff** [comment]
> So it can't be reproduced now after restarting APISIX?

No

**Comment by gxthrj** [comment]
I have tried use the cors with apisix admin-api, but got error:
```shell
curl http://127.0.0.1:9080/apisix/admin/routes/2 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -i -d '
{
    "uri": "/*",
    "hosts": ["foo.com"],
    "upstream": {
        "type": "roundrobin",
        "nodes": {
            "httpbin.org": 1
        }
    },
    "plugins": {
        "cors": {
         "allow_credential": true,
         "allow_headers": "*",
         "allow_methods": "*",
         "allow_origins": "*",
         "expose_headers": "*",
         "max_age": 5
        }
    }
}'
```

Got error as below

```shell
{"error_msg":"failed to check the configuration of plugin cors err: you can not set '*' for other option when 'allow_credential' is true"}
```

It seems that when the cors plug-in is created through the dashboard, there is no shcema verification about this.

**Comment by suifff** [comment]
> I have tried use the cors with apisix admin-api, but got error:
> 
> ```shell
> curl http://127.0.0.1:9080/apisix/admin/routes/2 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -i -d '
> {
>     "uri": "/*",
>     "hosts": ["foo.com"],
>     "upstream": {
>         "type": "roundrobin",
>         "nodes": {
>             "httpbin.org": 1
>         }
>     },
>     "plugins": {
>         "cors": {
>          "allow_credential": true,
>          "allow_headers": "*",
>          "allow_methods": "*",
>          "allow_origins": "*",
>          "expose_headers": "*",
>          "max_age": 5
>         }
>     }
> }'
> ```
> 
> Got error as below
> 
> ```shell
> {"error_msg":"failed to check the configuration of plugin cors err: you can not set '*' for other option when 'allow_credential' is true"}
> ```
> 
> It seems that when the cors plug-in is created through the dashboard, there is no shcema verification about this.

Thank you. I will modify this route through the apifix interface

**Comment by suifff** [comment]
> Can you provide a pasteable configuration so we can try it?
> Also, any interesting thing in the error.log?

There is no error log in the new request. When I restart, I found that there was an error log in apimix

![image](https://user-images.githubusercontent.com/37427008/120665381-4c6d6480-c4be-11eb-921c-fc0e39931d5e.png)

**Comment by starsz** [comment]
> I have tried use the cors with apisix admin-api, but got error:
> 
> ```shell
> curl http://127.0.0.1:9080/apisix/admin/routes/2 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -i -d '
> {
>     "uri": "/*",
>     "hosts": ["foo.com"],
>     "upstream": {
>         "type": "roundrobin",
>         "nodes": {
>             "httpbin.org": 1
>         }
>     },
>     "plugins": {
>         "cors": {
>          "allow_credential": true,
>          "allow_headers": "*",
>          "allow_methods": "*",
>          "allow_origins": "*",
>          "expose_headers": "*",
>          "max_age": 5
>         }
>     }
> }'
> ```
> 
> Got error as below
> 
> ```shell
> {"error_msg":"failed to check the configuration of plugin cors err: you can not set '*' for other option when 'allow_credential' is true"}
> ```
> 
> It seems that when the cors plug-in is created through the dashboard, there is no shcema verification about this.
> And APISIX will get error , when watch the change about cors plugin from etcd.
> 
> It is a bug in dashboard.
> 
> cc @nic-chen @starsz

In fact, the dashboard had schema verification about this :
you can referer: 

https://github.com/apache/apisix-dashboard/blob/9850df28b9539713566849069ba093b6be7c3be7/api/conf/schema.json#L2458

The error:
```
{"error_msg":"failed to check the configuration of plugin cors err: you can not set '*' for other option when 'allow_credential' is true"}
```
is validate by lua code.This can't be validated in apisix-dashboard.

you can referer to:
https://github.com/apache/apisix/blob/c61261af8d19557b77535c9c745f4f8182bb63b2/apisix/plugins/cors.lua#L134-L137

**Comment by nic-chen** [comment]
> The error:
```{"error_msg":"failed to check the configuration of plugin cors err: you can not set '*' for other option when 'allow_credential' is true"}```
is validate by lua code.This can't be validated in apisix-dashboard.

That's it.
